### PR TITLE
Add supported lexer validation to paste service and default the paste…

### PR DESCRIPTION
**Pull request by ChrisLovering**
…_url

### CODE REVIEW

File: pydis_core/utils/paste_service.py [hunk 1]
```diff
@@ -63,6 +623,18 @@ async def send_to_paste_service(
...
+            raise PasteUnsupportedLexerError(f"Lexer '{lexer}' not supported by pastebin.")
+
+        try:
+            async with http_session.get(f"{paste_url}/api/v1/lexer") as response:
+                response_json = await response.json()  # Supported lexers are the keys.
+        except Exception:
+            raise PasteUploadError("Could not fetch supported lexers from selected paste_url.")
+        if lexer not in response_json:
+            raise PasteUnsupportedLexerError(f"Lexer '{lexer}' not supported by pastebin.")
+
```
Comment (wookie184) [inline pydis_core/utils/paste_service.py:637]: Would it be more robust+simpler to just always send the request to get the lexers and cache that, so we don't need to keep the list of supported lexers up to date here? I doubt the extra request would be noticeable.

File: pydis_core/utils/paste_service.py [hunk 2]
```diff
@@ -63,6 +623,18 @@ async def send_to_paste_service(
...
     contents_size = len(contents.encode())
     if contents_size > max_size:
         log.info("Contents too large to send to paste service.")
```
Comment (wookie184) [inline pydis_core/utils/paste_service.py:640]: Not related to this PR, but given the discussion about actual size including syntax highlighting, isn't this logic now not sufficient to catch all pastes that are too large?

**Review by wookie184 — APPROVED** [review]
Looks good

File: pydis_core/utils/paste_service.py (hunk 1 above)
Comment (ChrisLovering) [reply pydis_core/utils/paste_service.py]: Good point, added in this forcepush https://github.com/python-discord/bot-core/compare/ed2c2fad86b8083ce82d7ceee47b4b3e1e256242..240683b9a0b28ea44a05f2740af143a69cf4ac0c

**Review by wookie184 — APPROVED** [review]

**Review by vivekashok1221 — APPROVED** [review]

**Pull request merged**